src/commodore/lieutenant.ts

```diff
@@ -0,0 +1,49 @@
+import normalizeUrl from 'normalize-url';
+
+import type { ClusterInfo } from './types';
+import { httpsQuery } from './https-query';
+import type { Response } from './https-query';
+
+type LieutenantErrorReply = {
+  reason: string;
+};
+
+export class LieutenantError extends Error {
+  statusCode: number;
+
+  constructor(message: string, statusCode: number) {
+    super(message);
+    Object.setPrototypeOf(this, LieutenantError.prototype);
+    this.name = this.constructor.name;
+    this.statusCode = statusCode;
+  }
+}
+
+export async function queryLieutenant(
+  config: any,
+  endpoint: string,
```

Comment:
We always query the `clusters` endpoint and we return `ClusterInfo`. Why is there even an option to specify another endpoint? The response won't make sense anyway when querying the `tenants` endpoint. 

I think it would make more sense to switch to something like `queryClusterInfo`?

Reply:
Yeah, that makes sense. I started out by trying to mimic Commodore's [`lieutenant_query()`](https://github.com/projectsyn/commodore/blob/b8dd895910208ced467dccf40031ccfe0bdecbcd/commodore/helpers.py#L95-L118), but in retrospect, that's completely unnecessary.

Reply:
Updated in latest version